ddl/ddl_worker.go

```diff
@@ -140,10 +140,21 @@ func newWorker(ctx context.Context, tp workerType, sessPool *sess.Pool, delRange
 		delRangeManager: delRangeMgr,
 	}
 	worker.addingDDLJobKey = addingDDLJobPrefix + worker.typeStr()
-	worker.logCtx = logutil.WithKeyValue(context.Background(), "worker", worker.String())
+	worker.logCtx = logutil.WithFields(context.Background(), zap.String("worker", worker.String()), zap.String("category", "ddl"))
 	return worker
 }
 
+func (w *worker) jobLogger(job *model.Job) *zap.Logger {
+	var traceInfo *model.TraceInfo
+	if job != nil {
+		traceInfo = job.TraceInfo
+	}
+	return logutil.LoggerWithTraceInfo(
+		logutil.Logger(w.logCtx).With(zap.Int64("jobID", job.ID)),
```

Comment:
We check `job != nil` in line149 before getting `traceInfo`, how about `job.ID`?

Reply:
done